### PR TITLE
Remove dealii compatibility fix

### PR DESCRIPTION
**Pull request by sibiaoliu**
Remove a not longer necessary compatibility fix for deal.ii 8.5.

### CODE REVIEW

File: source/simulator/core.cc [hunk 1]
```diff
@@ -797,23 +797,6 @@ namespace aspect
                 break;
               }
           }
-#else
```
Comment (bangerth) [inline source/simulator/core.cc]: You are removing both the `#else` and the `#endif`. You will then also have to remove the `#if` above.

Comment (sibiaoliu) [reply source/simulator/core.cc]: Removed.

**Comment by tjhei** [comment]
related to #2921

**Comment by sibiaoliu** [comment]
We merged the two commits into one.

**Pull request merged**